File: testDLLProjectCSHARP/testDLLProjectCSHARP/Class1.cs

using RGiesecke.DllExport;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;
using System.Threading.Tasks;
[ComVisible(true)]
public class Class1 {

    public String helloWorld() {
        return "Hello World C#!";
    }
}
static class UnmanagedExports {
    [DllExport]
    [return: MarshalAs(UnmanagedType.IDispatch)]
    static Object CreateTestClass() {
        return new Class1();
    }
}

File: testDllProject/testDllProject/Class1.vb
Imports System
Imports System.Collections.Generic
Imports System.Runtime.InteropServices
Imports System.Text
Imports System.Threading
Imports System.IO
Imports RGiesecke.DllExport

<ComVisible(True)>
Public Class Class1
    Public Function helloWorld() As String
        Return "HELLO WORLD!"
    End Function
End Class

Public Module UnmanagedExports
    <DllExport()> _
    Public Function CreateTestClass() As <MarshalAs(UnmanagedType.IDispatch)> Object
        Return New Class1
    End Function
End Module
